core/libp2p/transport/tcp/tcp_transport.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/transport/tcp/tcp_transport.hpp"
+
+namespace libp2p::transport {
+
+  void TcpTransport::dial(const multi::Multiaddress &address,
```

Comment:
Maybe, pass an address by value, so that it can be also constructed in-place when the client calls the method and don't need the address anymore? As you need to copy it anyway